pkg/controller/seed-controller-manager/cloud/cloud_controller.go

```diff
@@ -182,6 +185,7 @@ func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, clus
 		}
 	}
 
+	cluster.Spec.ComponentsOverride.Apiserver.NodePortRange = r.nodePortRange
```

Comment:
I'm not very fond of this. We should rely on the value we get from cluster spec and not override the value here. The best strategy to me would be to move the defaulting of `ComponentsOverride` to the mutating webhook.

Reply:
I've updated the PR to avoid making changes in cluster object